Trees/BinarySearchTree.hpp

```diff
@@ -0,0 +1,33 @@
+#ifndef BINARY_SEARCH_TREE
+#define BINARY_SEARCH_TREE
+
+struct Node {
+    int value;
+    struct Node * left_child;
+    struct Node * right_child;
+};
```

Comment:
There's no need to write `struct` when declaring an object of a structure in C++.

Therefore, change the last 2 lines to:
```C++
    Node* left_child;
    Node* right_child;
```

Also note that you should **not declare pointers like `type * name`!** Instead, you should either write `type* name` or `type *name`, though `type* name` is the preferred choice of most programmers. (Don't forget to make this change in other places as well.)